### ISSUE TITLE
TF Bound locations land within tall peaks in ATAC-seq data. Is this expected? 

### ISSUE THREAD

**Issue by rdbcasillas** [loosolab/TOBIAS#3]
Hello, 

Thank you for creating TOBIAS and making it user friendly. We are excited about the prospects and have some questions regarding its output on our ATAC seq datasets We ran TOBIAS to compare footprints for a transcription factor of interest in 2 conditions, and isolated regions where there are more footprints in condition A. When we started to visualize the footprints alongside the ATAC signal that went in, we notice for most of the genes footprints fall right within a clear tall peak. It's hard to see any conventional "notches" (regions with depleted reads) that are expected when looking for potential regions with footprints within ATAC signal. Are we looking in the wrong file? Can you share if this is something common you've observed as well? I understand that peaks are used to come up with footprints, which is maybe why footprints are within those regions but then what is the best way to visualize the actual location of these footprints? Do we need to correct it by few bases? 

Attached are the images that explain this concern. The track on the top is bigwig of ATACseq bam file. The one below shows the locations of bound TF. 

![footprint-1](https://user-images.githubusercontent.com/9725334/72569356-e576ed80-387f-11ea-9f1e-fcfd104e3db6.png)
![footprints-2](https://user-images.githubusercontent.com/9725334/72569358-e7d94780-387f-11ea-8fd0-9f3451d52286.png)
![image](https://user-images.githubusercontent.com/9725334/72569753-bca32800-3880-11ea-82b9-6e49b59b295c.png)
![footprints-zoomed](https://user-images.githubusercontent.com/9725334/72569435-12c39b80-3880-11ea-941e-b8545bfb2990.png)

Thank you!

**Comment by msbentsen** [comment]
Hi, thanks for getting in touch!

The footprint score from BigwigScores (~how likely a transcription factor is to bind) is a combined score taking into account the "notches"/depletions of the signal as well as the general accessibility (number of reads) around the binding site. Therefore, you can have a site be bound even if it does not show a visible "footprint" if the accessibility is high enough. 

For visualizing the possible footprints, I would load the "_corrected.bw" files coming from "TOBIAS ATACorrect". I don't know how you produced the bigwig that you show - is it the pileup of reads? The thing about footprints is that we need to look only at the Tn5 insertion site (one basepair signal), and not the whole read, as the read might extend into the footprint that we are trying to find. The ATACorrect tool will convert a .bam-file into this one basepair signal while correcting for Tn5-bias. Ranges of "negative" counts in this file will be representative of a "depletion" of signal due to protein binding - there is a small example [here](https://github.com/loosolab/TOBIAS/wiki/ATACorrect). 

Hope this explanation helps you out!
 
Best Mette

**Comment by rdbcasillas** [comment]
Thanks Mette, this helps. 

>I would load the "_corrected.bw" files coming from "TOBIAS ATACorrect". I don't know how you produced the bigwig that you show - is it the pileup of reads?

Yeah, I did that too and did notice things making slightly more sense visually but not really.  I simply created a bigwig using deeptools's bamCoverage normalization (1bp resolution) and loaded that. 

Attached is a picture that compares footprint output to the corrected BAM from ATACorrect. Below those two are the bound locations track. So according to TOBIAS, the region in the middle has highest footprint score because corresponding cut site signal is the lowest? I do see other regions in that image where I would have expected footprint score to be higher. But I guess what you are trying to show in the link you shared is that that's not how to visually think of footprint scores because they are calculated using mean flank  and mean mid, correct? 

<img width="1141" alt="Screenshot 2020-01-17 13 42 41" src="https://user-images.githubusercontent.com/9725334/72641631-3514f200-3930-11ea-9a56-207bbebcfe34.png">

Thanks again for helping me think through TOBIAS's output!

**Comment by msbentsen** [comment]
Yes, exactly - the footprint scores are not always fitting to the canonical idea of a footprint. This is mostly because the signal is so sparse, but also because there is a lot of randomness in the single Tn5 cuts, and in those cases we found it beneficial to also take accessibility into account. However, I will say that when you average the footprints over thousands of sites (as with TOBIAS PlotAggregate), the majority of transcription factors do show a footprinting effect! 

To sum up: the region in blue has a higher chance of binding due to the high accessibility of the flanks, even though the red region might seem to have more "depletion".
![72641631-3514f200-3930-11ea-9a56-207bbebcfe34_edit](https://user-images.githubusercontent.com/47495444/72710676-e3ef4300-3b67-11ea-99a8-5f364bfc9abd.png)

**Comment by rdbcasillas** [comment]
Thanks Mette, this discussion clarified some doubts and was really helpful.
